File: Structy/Introduction/Pairs.java
import java.util.List;
import java.util.ArrayList;

class Source {
  public static List<List<String>> pairs(List<String> elements) {
    List<List<String>> result = new ArrayList<>();

    for(int i = 0; i < elements.size(); i+=1){
      for(int j = i+1; j < elements.size(); j+=1){
        List<String> pair = List.of(elements.get(i), elements.get(j));
        result.add(pair);
      }
    }
    return result;
  }

  public static void run() {
    // this function behaves as `main()` for the 'run' command
    // you may sandbox in this function , but should not remove it
  }
}

/*
    n = length of elements
    Time: O(n^2)
    Space: O(n^2)
 */

File: Structy/Python/Introduction/Pairs.py
def pairs(elements):
  result = []
  
  for i in range(0,len(elements)):
    for j in range(i + 1, len(elements)):
      pair = [elements[i], elements[j]]
      result.append(pair)
      
  return result
# n is the length of the elements list
#Time Complexity: O(n^2)
#Space Complexity: O(n^2)
#Reason: We are using a list to store the result.